include/dali/core/geom/box.h

```diff
@@ -133,6 +133,22 @@ operator!=(const Box<ndims, CoordinateType> &lhs, const Box<ndims, CoordinateTyp
   return lhs.lo != rhs.lo || lhs.hi != rhs.hi;
 }
 
+
+template <int ndims, typename CoordinateType>
+std::ostream &operator<<(std::ostream &os, const Box<ndims, CoordinateType> &box) {
+  auto print_corner = [&os](const typename Box<ndims, CoordinateType>::corner_t &c) {
```

Comment:
Why don't you just use `operator<<` for `vec`? It has one.

Reply:
I'd replace this whole function with:
```
return os << box.lo << "-" << box.hi;
```
This would print sth lik:
```
{10, 10}-{630,470}
```

Reply:
Cause I think it's better to have it printed like `{(10, 10), (630, 470)}` and that's the format I wanted. So I couldn't use `operator<<` for `vec` :(